### ISSUE TITLE
`B017` has false negatives

### ISSUE THREAD

**Issue by ntBre** [astral-sh/ruff#19215]
[assert-raises-exception (B017)](https://docs.astral.sh/ruff/rules/assert-raises-exception/#assert-raises-exception-b017) has false negatives when the exceptions are provided as keyword arguments:

```python
import pytest
import unittest

with pytest.raises(Exception):  # true positive
    pass

with pytest.raises(expected_exception=Exception):  # false negative
    pass

class TestClass(unittest.TestCase):
    def test_okay(self):
        with self.assertRaises(Exception):  # true positive
            pass

    def test_false_negative(self):
        with self.assertRaises(exception=Exception):  # false negative
            pass
```

I think these also apply to the non-context-manager versions added in #19063. I have a feeling that there are other edge cases around argument validation here, but these are the examples I came up with as a start.

[playground](https://play.ruff.rs/e1ad5948-95f6-43bd-b687-b3dd95e6429f)

**Comment by danparizher** [comment]
I can grab this one since I worked on the previous issue
